src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,128 @@
+package duke;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileOutputStream;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+/**
+ * The Storage class deals with loading tasks from the file and saving tasks in the file.
+ */
+public class Storage {
+
+    private Path path;
+
+    /**
+     * Create and initialise a storage object.
+     * @param path The path where data are saved.
+     */
+    public Storage(Path path) {
+        this.path = path;
+    }
+
+    /**
+     * Return the list of task saved in file.
+     * @return list of task saved in file.
+     */
+    public List<Task> load() {
+        boolean directoryExists = java.nio.file.Files.exists(path);
+        List<Task> list = new ArrayList<>();
+
+        try {
+            if (directoryExists) {
+                FileReader fileReader = new FileReader(String.valueOf(path));
+                BufferedReader buffReader = new BufferedReader(fileReader);
+                while (buffReader.ready()) {
+                    String savedTask = buffReader.readLine();
+                    String type = savedTask.substring(0, 1);
+                    switch (type) {
+                    case "T":
+                        list.add(new ToDo(savedTask.substring(4).trim(), savedTask.substring(2, 3).equals("T")));
+                        break;
+                    case "D":
+                        String deadlineDetails = savedTask.substring(4);
+                        String[] deadlineArr = deadlineDetails.split("/by");
+                        Deadline deadline = new Deadline(
+                                deadlineArr[0].trim(),
+                                savedTask.substring(2, 3).equals("T"),
+                                LocalDateTime.parse(deadlineArr[1].trim()));
+                        list.add(deadline);
+                        break;
+                    case "E":
+                        String eventDetails = savedTask.substring(4);
+                        String[] eventArr = eventDetails.split("/at");
+                        Event event = new Event(
+                                eventArr[0].trim(),
+                                savedTask.substring(2, 3).equals("T"),
+                                LocalDateTime.parse(eventArr[1].trim()));
+                        list.add(event);
```

Comment:
Maybe this part can be abstracted into another method? like 1 method to read the file and another to convert the read data to actual objects.